### PR TITLE
Add switch to prevent backup cleanup

### PR DESCRIPTION
**Pull request by donomii**
### What's being changed:

Backup cleanup is failing on first run because permissions haven't propagated after cluster creation.  This adds an environment variable switch DISABLE_CLEANUP_UNFINISHED_BACKUPS to turn it off

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/rest/configure_api.go [hunk 1]
```diff
@@ -752,6 +752,9 @@ func configureAPI(api *operations.WeaviateAPI) http.Handler {
 	}
 	enterrors.GoWrapper(
 		func() {
+			// cleanup unfinished backups on startup
```
Comment (moogacs) [inline adapters/handlers/rest/configure_api.go:755]: **suggestion (if-minor):** could we make it configurable via env var, to be able to scale it up and down

File: adapters/handlers/rest/configure_api.go
Comment (moogacs) [inline adapters/handlers/rest/configure_api.go:1]: **thought:**   could we check for the permission every sec or so , so we don't have to wait a constant time and instead have feedback loop to sleep as much needed 🤔

File: adapters/handlers/rest/configure_api.go (hunk 1 above)
Comment (donomii) [reply adapters/handlers/rest/configure_api.go:755]: I don't understand, what would we be configuring?

File: adapters/handlers/rest/configure_api.go
Comment (donomii) [reply adapters/handlers/rest/configure_api.go:1]: Is that really worth it?  It's a background thread on clusters that are supposed to run without restarting for several weeks.

File: adapters/handlers/rest/configure_api.go [hunk 2]
```diff
@@ -750,12 +750,17 @@ func configureAPI(api *operations.WeaviateAPI) http.Handler {
 			}
 		}, appState.Logger)
 	}
-	enterrors.GoWrapper(
-		func() {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-			defer cancel()
-			backupScheduler.CleanupUnfinishedBackups(ctx)
-		}, appState.Logger)
+	suppressCleanup := os.Getenv("WEAVIATE_SUPPRESS_CLEANUP") == "true"
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:753]: can we use `entcfg.Enabled(...)` helper method to check if `WEAVIATE_SUPPRESS_CLEANUP` is set? and can we maybe rename it to `DISABLE_CLEANUP_UNFINISHED_BACKUPS`

File: adapters/handlers/rest/configure_api.go [hunk 3]
```diff
@@ -750,12 +751,15 @@ func configureAPI(api *operations.WeaviateAPI) http.Handler {
 			}
 		}, appState.Logger)
 	}
-	enterrors.GoWrapper(
-		func() {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-			defer cancel()
-			backupScheduler.CleanupUnfinishedBackups(ctx)
-		}, appState.Logger)
+	if !entcfg.Enabled("DISABLE_CLEANUP_UNFINISHED_BACKUPS") {
```
Comment (antas-marcin) [inline adapters/handlers/rest/configure_api.go:754]: This won't work... you have to change it to `entcfg.Enabled(os.Getenv("DISABLE_CLEANUP_UNFINISHED_BACKUPS"))`, you can checkout other places where we use this helper

**Comment by donomii** [comment]
redone in https://github.com/weaviate/weaviate/pull/6652

**Pull request closed without merging**